### PR TITLE
As/adaptive history autofill

### PR DESCRIPTION
**Pull request by soncuteanca**
# Description

Test to verify that typing the first three characters of a previously visited URL in the address bar triggers the adaptive history autofill.

## Bugzilla bug ID

**ID: 1902773**
**Link: https://bugzilla.mozilla.org/show_bug.cgi?id=1902773**

## Type of change

Please delete options that are not relevant.

- [x ] New Test

# How does this resolve / make progress on that bug?
- Completed

Please describe the progress or significance with respect to the bug listed above.

# Screenshots / Explanations

Please upload any relevant media or add a relevant description with respect to the bug listed above.

# Comments / Concerns

Please add a short blurb about any comments or concerns that this change might cause.

### CODE REVIEW

File: tests/address_bar_and_search/test_adaptive_history_autofill.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    # Type the first 3 characters of the visited URL in the address bar and select the suggested URL
+    nav.type_in_awesome_bar("new")
+    nav.get_element("firefox-suggest").click()
+    nav.expect_in_content(
+        EC.url_contains("https://news.google.com/home?hl=en-US&gl=US&ceid=US:en")
+    )
+
+    # Open a new tab, type the first 3 characters of the visited URL and see that it is autofilled directly
+    tabs.new_tab_by_button()
```
Comment (ben-c-at-moz) [inline tests/address_bar_and_search/test_adaptive_history_autofill.py:48]: we need to switch to the new tab before we can confirm this worked:
```python
tabs.wait_for_num_tabs(2)
driver.switch_to.window(driver.window_handles[-1])
```

**Review by ben-c-at-moz — CHANGES_REQUESTED** [review]
One small thing, the rest looks great!

File: tests/address_bar_and_search/test_adaptive_history_autofill.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        x_icon = tabs.get_element("tab-x-icon", multiple=True)
+        x_icon[0].click()
+
+    # Type the first 3 characters of the visited URL in the address bar and select the suggested URL
+    nav.type_in_awesome_bar("nat")
+    nav.get_element("firefox-suggest").click()
+    nav.expect_in_content(
+        EC.url_contains("https://www.nationalgeographic.com/science/")
+    )
+
```
Comment (emilghittasv) [inline tests/address_bar_and_search/test_adaptive_history_autofill.py:46]: Add `tabs.set_content_context()` to switch context back to content.

**Review by emilghittasv — CHANGES_REQUESTED** [review]

**Review by ben-c-at-moz — APPROVED** [review]
Looks good!

**Review by emilghittasv — APPROVED** [review]
approved!

**Pull request merged**